test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -864,6 +864,80 @@ main() {
       });
     });
 
+    group('Dart 1 compatibility', () {
+      test('ignore line is not removed if expected part directive does not follow', () {
+        final ignoreLine = '\/\/ ignore: uri_does_not_exist, uri_has_not_been_generated';
+
+        setUpAndGenerate('''
+            // ignore: uri_does_not_exist, uri_has_not_been_generated
+            part 'component.dart';
+
+            @Factory()
+            UiFactory<FooProps> Foo;
+
+            @Props()
+            class FooProps {}
+
+            @Component()
+            class FooComponent {
+              render() => null;
+            }
+          ''');
+
+        var transformedSource = transformedFile.getTransformedText();
+
+        expect(transformedSource, contains(ignoreLine));
+      });
+
+
+      test('ignore line and part directive is removed from component file', () {
+        final ignoreLine = '\/\/ ignore: uri_does_not_exist, uri_has_not_been_generated';
+        final partLine = "part \'component\.overReact\.g\.dart\';";
```

Comment:
Needs one more test to verify that the part directive is removed even if the ignore comment isn't found